lib/src/page_list_viewport_gestures.dart

```diff
@@ -528,3 +772,77 @@ class PanningFrictionSimulation implements PanningSimulation {
 
   bool isDone(double time) => _xSimulation.isDone(time) && _ySimulation.isDone(time);
 }
+
+class FrictionAndFirstOrderDragMomentumSimulation extends Simulation {
+  FrictionAndFirstOrderDragMomentumSimulation(
+      double friction, double drag, double mass, double position, double velocity, double scalar,
+      {super.tolerance, double initialVelocityScalar = 1, double maxScrollingVelocity = 100000})
+      : _c = drag,
+        _n = friction,
+        _m = mass,
+        _x = position,
+        _w = velocity.abs() * initialVelocityScalar,
+        _sign = velocity.sign,
+        _scalar = scalar {
+    _finalTime = _m * math.log(1 + _w * _c / (_m * _n)) / _c;
+    if (_w > maxScrollingVelocity) {
+      _w = maxScrollingVelocity;
+    }
+  }
+
+  final double _c; // Fluid drag first order
+  final double _n; // Static friction
+  final double _x; // Initial position
+  final double _m; // Mass
+  double _w; // Absolute value of the initial velocity
+  final double _sign; // Sign of the initial velocity
+  // Scalar, by which to multiply all the positional results.
+  // Needed to calculate projection onto an arbitrary simulation axis.
+  final double _scalar;
+  double _finalTime = double.infinity; // Total time for the simulation, initialized upon build
+
+  @override
+  double x(double time) {
+    if (time > _finalTime) {
+      return finalX;
+    }
+    // Computes the position `x` at time `t`:
+    // x = \frac{\left(k_{1}m\ e^{\frac{-c\ t}{m}}-m\ n\ t\right)}{c}+k_{2}
```

Comment:
What are these comments? Is this code? If so, please don't commit commented code.

Reply:
This is LaTeX. This is helpful to format math. Most plotting software supports this syntax by default or has a convenient interpreter. For example if you just paste this into desmos, you'll be able to play with the curves yourself.